File: app/src/main/java/com/example/learn/ui/partition/partition/one/OneView.java
package com.example.learn.ui.partition.partition.one;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.learn.R;
import com.example.learn.ui.partition.IPartitionCallback;

public class OneView extends LinearLayout {

    private TextView oneViewText;

    public OneView(@NonNull Context context) {
        super(context);
        initView(context);
    }

    public OneView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        initView(context);
    }

    public OneView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initView(context);
    }

    private void initView(Context context) {
        //
        inflate(context, R.layout.one_view, this);
        //
        oneViewText = (TextView) findViewById(R.id.one_view_text);
        //
        setOrientation(VERTICAL);
    }

    public void initData(final OneBean.DataBean dataBean, final IPartitionCallback callback, final int position) {
        if (dataBean == null || callback == null) {
            return;
        }
        oneViewText.setText(dataBean.getText());
        oneViewText.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                callback.clickOne(position, dataBean.getText());
            }
        });
    }
}

File: app/src/main/java/com/example/learn/ui/partition/partition/one/OneView.kt
package com.example.learn.ui.partition.partition.one

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.LinearLayout
import com.example.learn.R
import com.example.learn.ui.partition.IPartitionCallback
import kotlinx.android.synthetic.main.one_view.view.*

class OneView @JvmOverloads constructor(
    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
) : LinearLayout(context, attrs, defStyleAttr) {

    init {
        //
        View.inflate(context, R.layout.one_view, this)
        //
        orientation = VERTICAL
    }

    fun initData(dataBean: OneBean.DataBean?, callback: IPartitionCallback?, position: Int) {
        if (dataBean == null) {
            return
        }
        oneViewText.text = dataBean.text
        oneViewText.setOnClickListener { callback?.clickOne(position, dataBean.text) }
    }
}